src/components/UpdateAvatar.js

```diff
@@ -0,0 +1,42 @@
+import Popup from "./Popup.js";
+import api from "./Api.js"; // Import the api instance
+
+class UpdateAvatar extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._profileImageInput = this._popupElement.querySelector(
+      "#profile-image-input"
+    );
+    this._profileImageButton = document.querySelector(".profile__image-button");
+    this._handleFormSubmit = this._handleFormSubmit.bind(this);
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    const profileImageForm = this._popupElement.querySelector(
+      "#profile-image-form"
+    );
+    profileImageForm.addEventListener("submit", this._handleFormSubmit);
+    this._profileImageButton.addEventListener("click", () => {
+      this.open();
+    });
+  }
+
+  _handleFormSubmit(event) {
+    event.preventDefault();
+    const newImageUrl = this._profileImageInput.value;
+
+    api
```

Comment:
Needs correction: please don't use api inside the classes, you can use api only in index.js